forte/data/data_store.py

```diff
@@ -258,18 +276,28 @@ def __getstate__(self):
         """
         state = super().__getstate__()
         state["_DataStore__elements"] = {}
+
+        # Make a copy of the updated type_attributes
+        type_attributes = deepcopy(DataStore._type_attributes)
```

Comment:
why do we need a deepcopy?

Reply:
It is because we want to remove only the type information from the type_attributes dict before serialization. However, since a dict is mutable in python, we cannot do this operation on the original `_type_attributes` dict. Thus, we create a copy of this dictionary, remove the type information from that and then use it for serialization. (Here, type information mainly means the dictionary stored by key `type` in the `_type_attributes` for each entry)

Reply:
It is because we want to only remove the type information atom the `_type_attributes` before serialization without affecting the class variable. Since `dict` is mutable, we create a copy of `_type_attributes` which will be modified and then used during serialization.

Reply:
so this variable will go to serialized data. I normally prefer the other way around where we assemble the serialized data instead of removing stuff from a dict. One reason is that we have more contorl over the data being serialized

Reply:
Yes I actually had started off with your approach, but I noticed that it made the code a bit large unnecessarily because the part to remove (type information) is much lesser than the part to keep. But if you want I can try the other way round as well.

Reply:
ok, btw. is it necessary to remove the type information?

Reply:
Yes, because they are not stored in serializable formats